test/drenv/envfile.py

```diff
@@ -83,20 +88,24 @@ def _validate_profile(profile):
     profile.setdefault("workers", [])
 
     for i, worker in enumerate(profile["workers"]):
-        _validate_worker(worker, profile, i)
+        _validate_worker(worker, profile, addons_dir, i)
 
 
-def _validate_worker(worker, env, index):
+def _validate_worker(worker, env, addons_dir, index):
     worker["name"] = f'{env["name"]}/{worker.get("name", index)}'
     worker.setdefault("addons", [])
 
     for addon in worker["addons"]:
-        _validate_addon(addon, env, args=[env["name"]])
+        _validate_addon(addon, env, addons_dir, args=[env["name"]])
 
 
-def _validate_addon(addon, env, args=()):
+def _validate_addon(addon, env, addons_dir, args=()):
     if "name" not in addon:
-        raise ValueError(f"Missing addon 'name': {addon}")
+        raise ValueError("Missing name")
+
+    addons_dir = os.path.join(addons_dir, addon["name"])
```

Comment:
this is `addon_dir` and not `addons_dir`, which changes the input argument. Changing the input argument is typically bad idea and will break stuff later even if it works now.

Having both `addons_dir` and `addon_dir` is not great, the names are too similar. Maybe we need a different name for `addons_dir`, like `addons_root` to avoid this confusion?